velox/expression/SimpleFunctionAdapter.h

```diff
@@ -270,6 +288,9 @@ class SimpleFunctionAdapter : public VectorFunction {
       ApplyContext& applyContext,
       bool allNotNull,
       const TReader&... readers) const {
+    bool callNullFree = FUNC::is_default_contains_nulls_behavior ||
```

Comment:
1. I assume FUNC::is_default_contains_nulls_behavior implies FUNC::udf_has_callNullFree right?
2. We call callNullFree if the FUNC::udf_has_callNullFree and !applyContext.mayHaveNullsRecursive
otherwise we either return null or call (call, or callNullable). 
shall we make it . 
bool callNullFree = FUNC::udf_has_callNullFree && !applyContext.mayHaveNullsRecursive; 
and add assert that if FUNC::is_default_contains_nulls_behavior is true, FUNC::udf_has_callNullFree should be true.